manifests/ctrl_alt_del.pp

```diff
@@ -0,0 +1,110 @@
+# Manage the state of pressing ``ctrl-alt-del``
+#
+# @param enable
+#   Allow ``ctrl-alt-del`` to restart the system
+#
+# @param log
+#   Instead of just disabling the command, set the system up to write a log
+#   entry when the key combination is pressed
+#
+# @param log_users
+#   Record all logged in users in the log message
+#
+# @author Trevor Vaughan <tvaughan@onyxpoint.com>
+#
+class simp::ctrl_alt_del (
+  Boolean                   $enable    = false,
+  Boolean                   $log       = true,
+  Boolean                   $log_users = true,
+  Simplib::Syslog::Facility $facility  = 'local6',
+  Simplib::Syslog::Severity $severity  = 'warning'
+) {
+
+  if 'systemd' in $facts['init_systems'] {
+    $_logger = '/bin/echo -n'
+  }
+  else {
+    $_logger = "/bin/logger -p ${facility}.${severity}"
+  }
+
+  if $log {
+    if $log_users {
+      $_log_cmnd = "/bin/sh -c \"${_logger} 'Ctrl-Alt-Del detected - Logged in users:' `/usr/bin/who | /bin/cut -f1 -d' ' | /bin/sort -u | /usr/bin/tr '\\n' ' '`\""
+    }
+    else {
+      $_log_cmnd = "/bin/sh -c \"${_logger} 'Ctrl-Alt-Del detected'\""
+    }
+  }
+
+  if 'systemd' in $facts['init_systems'] {
+    if $enable {
```

Comment:
I can't tell if the $enable param does what I expect or not. I understand that true leaves the default behavior, but it disables the feature of this class.